src/OpenTelemetry/Trace/Export/Internal/BroadcastActivityProcessor.cs

```diff
@@ -84,6 +84,17 @@ public override Task ShutdownAsync(CancellationToken cancellationToken)
             return Task.WhenAll(tasks);
         }
 
+        public override Task ForceFlushAsync(CancellationToken cancellationToken)
+        {
+            var tasks = new List<Task>();
+            foreach (var processor in this.processors)
+            {
+                tasks.Add(processor.ForceFlushAsync(cancellationToken));
+            }
+
+            return Task.WhenAll(tasks);
```

Comment:
We have this pattern in other locations but this needs to terminate in case of cancellation (suppose a buggy processor that doesn't respect the cancellation). That said we should address this separately not on this PR. The typical pattern is to register a TaskCompletionSource with the token and use properly combine it with WhenAny and WhenAll.